index.js

```diff
@@ -622,14 +622,14 @@ function base64Slice (buf, start, end) {
   }
 }
 
-function decodeCodePointsArray (array) {
-  // Based on http://stackoverflow.com/a/22747272/680742, the browser with
-  // the lowest argument limit is Chrome, with 0x10000 args.
-  if (array.length < 0x10000) return String.fromCharCode.apply(String, array)
+function decodeCodePointsArray (buf) {
+  var len = buf.length
+
+  if (len <= 0x10000) {
```

Comment:
The issue isn't manual string concatenation; it's doing an unnecessary `Array.slice()` when the buffer.length is < 0x10000. I think it's pretty clear that doing `slice()` is slower than not doing `slice()`, although a JSPerf would tell us exactly how slow.